http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/nrs/HttpStreamsHandler.java

```diff
@@ -211,7 +215,8 @@ private void handleCancelled(ChannelHandlerContext ctx, InT msg) {
     private void handleReadHttpContent(ChannelHandlerContext ctx, HttpContent content) {
         if (!ignoreBodyRead) {
             if (content instanceof LastHttpContent) {
-
+                ctx.channel().attr(LAST_HTTP_CONTENT_RECEIVED_KEY).set(true);
```

Comment:
Do we know in what situations `ignoreBodyRead` is set to true?

Reply:
`ignoreBodyRead` is used in cases like Http 200 OK message cases where the inMsg is empty and we are not expecting any more bytes to be read.

https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/nrs/HttpStreamsHandler.java#L160